src/ProjNet/CoordinateSystems/CompoundCoordinateSystem.cs

```diff
@@ -0,0 +1,109 @@
+                }
+                sb.Append("]");
+                return sb.ToString();
+            }
+        }
+
+        /// <inheritdoc/>
+        public override string XML
+        {
+            get
+            {
+                var sb = new StringBuilder();
+                sb.AppendFormat(CultureInfo.InvariantCulture.NumberFormat,
+                    "<CS_CoordinateSystem Dimension=\"{0}\"><CS_CompoundCoordinateSystem>{1}",
+                    this.Dimension, InfoXml);
+                foreach (var ai in AxisInfo)
+                    sb.Append(ai.XML);
+                sb.Append(HeadCoordinateSystem.XML);
+                sb.Append(TailCoordinateSystem.XML);
+                sb.AppendFormat("</CS_CompoundCoordinateSystem></CS_CoordinateSystem>");
+                return sb.ToString();
+            }
+        }
+
+        /// <summary>
+        /// The head coordinate system
+        /// </summary>
+        public CoordinateSystem HeadCoordinateSystem { get => _headCoordinateSystem; set { _headCoordinateSystem = value; } }
+
+        /// <summary>
+        /// The tail coordinate system
+        /// </summary>
+        public CoordinateSystem TailCoordinateSystem { get => _tailCoordinateSystem; set { _tailCoordinateSystem = value; } }
+        /// <summary>
+        /// A compound coordinate system
+        /// </summary>
+        /// <param name="headcs">The head (first) coordinate system</param>
+        /// <param name="tailcs">The tail (second) coordinate system</param>
+        /// <param name="name">Name</param>
+        /// <param name="authority">Authority name</param>
+        /// <param name="authorityCode">Authority-specific identification code</param>
+        /// <param name="alias">Alias</param>
+        /// <param name="abbreviation">Abbreviation</param>
+        /// <param name="remarks">Optional information</param>
+        public CompoundCoordinateSystem(CoordinateSystem headcs, CoordinateSystem tailcs, string name, string authority, long authorityCode, string alias, string abbreviation, string remarks)
+            : base(name, authority, authorityCode, alias, abbreviation, remarks)
+        {
+            _headCoordinateSystem = headcs;
+            _tailCoordinateSystem = tailcs;
+            AxisInfo = new List<AxisInfo>();
+            AxisInfo.AddRange(HeadCoordinateSystem.AxisInfo);
+            AxisInfo.AddRange(TailCoordinateSystem.AxisInfo);
+        }
+
+        /// <inheritdoc/>
+        public override bool EqualParams(object obj)
+        {
+            if( obj is CompoundCoordinateSystem compdCs )
+            {
+                return HeadCoordinateSystem.EqualParams(compdCs.HeadCoordinateSystem) && TailCoordinateSystem.EqualParams(compdCs.TailCoordinateSystem);
+            }
+
+            return false;
+        }
+
+        /// <inheritdoc/>
+        public override IUnit GetUnits(int dimension)
+        {
+            if( dimension < 0 || dimension >= Dimension )
```

Comment:
Shouldn't we override Dimension, too, or where is it set?

Reply:
Dimension is just a getter that returns the count of AxisInfo objects in the coordinate system. So by setting the AxisInfo to the combination of the two coordinate system's axes in the constructor, we set the dimension.

Reply:
Ok.